### PR TITLE
Refactor git storage and command

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: modules/git/command.go [hunk 1]
```diff
@@ -40,19 +40,17 @@ const DefaultLocale = "C"
 
 // Command represents a command with its subcommands or arguments.
 type Command struct {
-	name             string
 	args             []string
 	parentContext    context.Context
 	desc             string
+	extraDesc        string
```
Comment (wxiaoguang) [inline modules/git/command.go:46]: Why?

Comment (lunny) [reply modules/git/command.go:46]: To keep the default description, so we can remove most descriptions when invoking git command.

Comment (wxiaoguang) [reply modules/git/command.go:46]: Then `SetDescription` will be removed, right?

Comment (lunny) [reply modules/git/command.go:46]: > 

That still will be kept but in most places, we don't need that function.

Comment (wxiaoguang) [reply modules/git/command.go:46]: > SetDescription still will be kept but in most places, we don't need that function.

Then I still can not get the point why `SetDescription` and `SetExtraDescription` should both exist.

Comment (wxiaoguang) [reply modules/git/command.go:46]: After grepping the calls to `SetDescription` and `SetExtraDescription`. If I understand correctly, the `SetDescription` is mainly for making the description of git command with long arguments to be concise and add repoPath to the description.

But `SetDescription` and `SetExtraDescription` really seem similar and may be misused later ....

(just a discussion, no more thoughts from my side)

Comment (wxiaoguang) [reply modules/git/command.go:46]: Just thought about an interesting idea:

`cmd.SetDescription("totally new description: %s, %d", filename, line)`
`cmd.SetDescription("${cmd}, and extra description: %s, %d", filename, line)`

The `${cmd}` could be replaced by the command and arguments.

Then only one function is enough. And one command always have one description (no `extra`). And no need to call `Sprintf` separately when setting the description.


ps: I know the trick `${cmd}` is arguable, so not sure whether it is good enough, just FYI and for fun.

**Comment by lunny** [comment]
replaced by #25071

**Pull request closed without merging**